src/modules/75-cf/pages/feature-flags/__tests__/FeatureFlagsPage.test.tsx

```diff
@@ -220,6 +230,52 @@ describe('FeatureFlagsPage', () => {
     expect(getByText(document.body, message)).toBeDefined()
   })
 
+  test('It should go to Feature Flag details page on click of an environment in All Environments view', async () => {
+    const refetchProjectFlags = jest.fn()
+
+    mockImport('services/cf', {
+      useGetProjectFlags: () => ({
+        loading: false,
+        data: mockGetAllEnvironmentsFlags,
+        refetch: refetchProjectFlags,
+        error: null
+      })
+    })
+
+    renderComponent()
+
+    const environmentSelect = screen.getByRole('textbox', { name: 'cf.shared.selectEnvironment' })
+
+    expect(environmentSelect).toHaveValue('foobar')
+
+    userEvent.click(environmentSelect)
+
+    expect(refetchProjectFlags).not.toHaveBeenCalled()
+    expect(screen.getByText('common.allEnvironments')).toBeInTheDocument()
+
+    userEvent.click(screen.getByText('common.allEnvironments'))
+
+    expect(refetchProjectFlags).toHaveBeenCalledTimes(1)
+    expect(screen.getAllByText('cf.environments.nonProd')).toHaveLength(17)
+    expect(screen.getAllByText('cf.environments.prod')).toHaveLength(17)
+
+    const rows = screen.getAllByRole('row')
+
+    // Feature Flag 1st row
+    const flag1Columns = getAllByRole(rows[1], 'cell')
```

Comment:
Nothing wrong here, but just wanted to let you know that `within` exists and is great for this kinda thing. Both do the same thing in this case, so no need to change, just an FYI
```suggestion
    const flag1Columns = within(rows[1]).getAllByRole('cell')
```

Reads better IMO